src/Microsoft.Framework.Runtime.Compilation.DesignTime/DesignTimeHostCompiler.cs

```diff
@@ -76,8 +76,7 @@ private void OnError(int? contextId, string error)
             int contextId;
             if (!contexts.TryGetValue(projectPath, out contextId))
             {
-                // This should never happen
-                throw new InvalidOperationException();
+                throw new InvalidOperationException($"The context of the project at {projectPath} could not be determined. It could be caused by multiple global.json files specifying different versions of the SDK for the same project.");
```

Comment:
The problem happens if you have multiple global.json (which I understand can only happen if you are using source references) and the sdk versions in these are different. In this case VS will start multiple dnx.exe processes (one for each sdk version). When building the project VS will use the dnx.exe corresponding to the sdk version but since it is a single project building will fail because the part of the project built in one of the dnx.exe will not be able to find references to part of the project built in other dnx.exe